src/openms/include/OpenMS/ANALYSIS/OPENSWATH/OpenSwathOSWWriter.h

```diff
@@ -392,23 +392,50 @@ namespace OpenMS
           std::vector<double> id_target_ind_xcorr_shape = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_xcorr_shape"),';');
           std::vector<double> id_target_ind_log_sn_score = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_log_sn_score"),';');
           std::vector<double> id_target_ind_massdev_score = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_massdev_score"),';');
-          std::vector<double> id_target_ind_mi_score = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_mi_score"),';');
-          std::vector<double> id_target_ind_mi_ratio_score = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_mi_ratio_score"),';');
           std::vector<double> id_target_ind_isotope_correlation = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_isotope_correlation"),';');
           std::vector<double> id_target_ind_isotope_overlap = ListUtils::create<double>((String)feature_it->getMetaValue("id_target_ind_isotope_overlap"),';');
 
           if ((String)feature_it->getMetaValue("id_target_num_transitions") != "")
           {
-            for (int i = 0; i < feature_it->getMetaValue("id_target_num_transitions").toString().toInt(); ++i)
+            int id_target_num_transitions = feature_it->getMetaValue("id_target_num_transitions").toString().toInt();
+
+            std::vector<std::string> id_target_total_mi(id_target_num_transitions, "NULL");
+            if (!feature_it->getMetaValue("id_target_total_mi").isEmpty())
+            {
+              id_target_total_mi = ListUtils::create<std::string>((String)feature_it->getMetaValue("id_target_total_mi"),';');
+            }
+
+            std::vector<std::string> id_target_ind_mi_score(id_target_num_transitions, "NULL");
+            if (!feature_it->getMetaValue("id_target_ind_mi_score").isEmpty() && feature_it->getMetaValue("id_target_ind_mi_score") != "")
             {
+              id_target_ind_mi_score = ListUtils::create<std::string>((String)feature_it->getMetaValue("id_target_ind_mi_score"),';');
+            }
+
+            std::vector<std::string> id_target_ind_mi_ratio_score(id_target_num_transitions, "NULL");
+            if (!feature_it->getMetaValue("id_target_ind_mi_ratio_score").isEmpty())
+            {
+              id_target_ind_mi_ratio_score = ListUtils::create<std::string>((String)feature_it->getMetaValue("id_target_ind_mi_ratio_score"),';');
+            }
+
+            for (int i = 0; i < id_target_num_transitions; ++i)
+            {
+              if (id_target_total_mi[i] == "") {
+                id_target_total_mi[i] = "NULL";
+              }
+              if (id_target_ind_mi_score[i] == "") {
+                id_target_ind_mi_score[i] = "NULL";
+              }
+              if (id_target_ind_mi_ratio_score[i] == "") {
+                id_target_ind_mi_ratio_score[i] = "NULL";
```

Comment:
why do you expect empty strings? does OpenSWATH write out empty strings?

Reply:
PS: also can you change the style to
```
if ... 
{
}
```

Reply:
The transition-level scores are still reported as a concatenated meta feature (e.g. "2;3;4"). If (some of) the MI scores are disabled, an empty string might be reported. This part might be worthwhile to refactor in the future.

I changed the style accordingly.